app.rb

```diff
@@ -1,20 +1,226 @@
+#   puts apartment_1_tenant
+# puts "*" * 50
+#
+#   # Print all the tenants in order from youngest to oldest
+#   tenants_by_age = tenants.sort_by { |e| e[:age]  }
+#   puts tenants_by_age
+#
+#   To display oldest to youngest:
+#   puts tenants_by_age.reverse
+#
+# puts "*" * 50
+#
+#   # Print the names of all the tenants alphabetically
+# puts "Tell your tenants to stop adding junk to their names."
+# puts "*" * 50
+#
+#   ## More challenging
+#
+#   # When printing tenants also print out the address that the tenant resides in.
+# # tenants_address = tenants.map do |n|
+# #   apartments[address]
+# # end
+# tenants_address = []
+#  tenants.inject([]) do |apartments, address|
+#  tenants_address  <<  [:address]
+# end
+#         # ^^^^^ This doesn't work ^^^^
+# Try this:
+# puts tenants_address
+#
+# Or this:
+# tenants.each do |tenant|
+#   puts tenant
+#   apartment_id = tenant[:apartment_id]
+#
+#   # When printing all apartments, under each apartment print all of its tenants
+# ## More challenging
+# When printing tenants also print out the address that the tenant
+# resides in.
+#
+# tenants.each do |tenant|
+#
+#   tenants_apartment_id = tenant[:apartment_id]
+#   matching_apartment = apartments.find do |apartment|
+#     apartment[:id] == tenants_apartment_id
+#   end
+#
+#   puts "#{tenant[:name]} lives at #{matching_apartment[:address]}"
+#   puts "*" * 50
+# end
+#
+#
+# # When printing all apartments, under each apartment print all of its tenants
+# # apartments.each do |apartment|
+# #   puts apartment[:address]
+# #   tenants_of_apartment = tenants.select{|tenant| tenant[:apartment_id] == apartment[:id]}
+# #   tenants_of_apartment.each do |tenant|
+# #     puts tenant[:name]
+# #   end
+# #   puts nil # empty line
+# # end
+# #   Bob Pizza is old school, and he wants a sick command line interface for this app. Bob's user stories:
+# #
+# #   ### MVP:
+# #   ```
+# #   ```
+# #
+#
+# #   - I should have a numbered interface for my application so that I can just type in a number to access different parts of my program.
+# puts "Starting review.  Make sure your Terminal window is at least 90 columns wide."
+# puts "Choose option to view data:"
+# puts "(1) View All Apartments"
+# puts "(2) View All Tenants"
+# puts "(3) View Tenants in each Apartment"
+# puts "(4) Search For a Tenant and Get Address"
+# puts "(x) to Exit"
+#
+# input = gets.chomp
+#
+# if input == "x"
+#   exit
+# #   - I should be allowed an option to view all the apartments so that I can get an overview of my properties.
+# elsif input == "1"
+#   puts apartments
+#
+#   #   - I should be allowed  an option view all the tenants so that I can get an overview of my tenants
+# elsif input == "2"
+#   puts tenants
+#
+#   #   - I should be allowed an option to view all the apartments with their tenants so that I can get an overview of Occupancy.
+# elsif input == "3"
+#   puts tenants.sort_by { |tenant| tenant[:apartment_id]  }
+#   puts "*" *20
```

Comment:
I think you have a lot of good code here. One issue that I can see, is that it will only happen once. How can we leverage a loop to continue running this logic un the user types `x`